tests/integration/construct_component_methods_unknown/testcomponent-python/testcomponent.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2016-2021, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+import sys
+
+import pulumi
+import pulumi.provider as provider
+
+
+def panic(text: str):
+    print(text)
+    sys.exit(1)
+
+
+class Component(pulumi.ComponentResource):
+    def __init__(self,
+                 resource_name: str,
+                 opts: Optional[pulumi.ResourceOptions] = None) -> None:
+        super().__init__("testcomponent:index:Component", resource_name, {}, opts)
+
+    def get_message(self, echo: pulumi.Input[str]) -> pulumi.Output[str]:
+        message = pulumi.Output.from_input(echo)
+        message.apply(lambda v: panic("should not run (echo)"))
+        return message
```

Comment:
Could make it a bit stronger by `return  message.apply(lambda v: panic("should not run (echo)"))`? So we don't rely on "we wait for all outstanding applies". I should perhaps not be skeptical that it works, but I've just been chasing bugs in that in Go SDK.

Reply:
Done